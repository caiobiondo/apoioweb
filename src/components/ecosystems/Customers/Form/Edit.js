import CustomerForm from './Form';

class EditCustomerForm extends CustomerForm {
  constructor(props) {
    super(props);
    const { id } = this.props.match.params;
    this.customer = { id: id };
    this.getCustomer = this.getCustomer.bind(this);
  }

  getCustomer() {
    return this.customer;
  }
}

export default EditCustomerForm;
