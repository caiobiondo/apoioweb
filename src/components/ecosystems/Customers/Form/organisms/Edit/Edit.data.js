import gql from 'graphql-tag';

export const UpdateCustomerMutation = gql`
  mutation UpdateCustomer($input: UpdateCustomerInput!) {
    updateCustomer(input: $input) {
      customer {
        id
        name
        nickname
      }
    }
  }
`;

export const FindCustomerQuery = gql`
  query FindCustomerQuery($customerId: Int!) {
    customer(customerId: $customerId) {
      id
      nickname
      name
      gender
      comment
      birthday
      emails {
        id
        email
      }
      phones {
        id
        phone
        provider
      }
      addresses {
        id
        additional_address
        city
        neighborhood
        street_name
        street_number
        state
        zipcode
      }
    }
  }
`;

export const FindCustomerQueryOptions = {
  options(props) {
    return {
      cachePolicy: 'no-cache',
      forceFetch: true,
      variables: {
        customerId: props.match.params.id,
      },
    };
  },
  props({ data: { loading, customer } }) {
    return {
      loading,
      customer,
    };
  },
};
