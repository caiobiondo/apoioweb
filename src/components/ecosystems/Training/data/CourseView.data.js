import gql from 'graphql-tag';
import { Origem } from 'config';

export const CourseViewQuery = gql`
  query CourseViewQuery(
    $sellerId: Int!
    $courseId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    course(
      sellerId: $sellerId
      courseId: $courseId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      id
      accessToken
      categoryTitle
      clientIdentifier
      clientSecrets
      courseContent {
        video
        html5
        web
        videoEmbedUrl
        scorm
      }
      dateUpload
      durationInSeconds
      description
      generalRating
      isfavorite
      myRating
      ratedByYou
      relatedCourses {
        id
        title
        thumbnail
        type
        url
      }
      status
      stoppedAt
      thumbnail
      title
      type
      views
    }
  }
`;

export const CourseViewQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        courseId: props.courseId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'cache-and-network',
      //fetchPolicy: 'no-cache',
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      course: data.course,
      refetch: data.refetch,
    };
  },
};
