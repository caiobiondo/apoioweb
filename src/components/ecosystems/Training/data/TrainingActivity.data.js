import gql from 'graphql-tag';
import { Origem } from 'config';

export const ActivityViewQuery = gql`
  query ActivityViewQuery(
    $sellerId: Int!
    $activityId: Int!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    activity(
      sellerId: $sellerId
      activityId: $activityId
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      status
      message
      totalOfElement
      results {
        id
        finished
        name
        type
        accessToken
        courseContent {
          video
          html5
          web
          videoEmbedUrl
          html5Embed
          scorm
        }
        assessment {
          id
          title
          questions {
            id
            description
            type
            alternatives {
              id
              description
            }
          }
        }
      }
    }
  }
`;

export const ActivityViewQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        activityId: props.activityId,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'network-only',
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      activity: data.activity,
      refetch: data.refetch,
    };
  },
};
