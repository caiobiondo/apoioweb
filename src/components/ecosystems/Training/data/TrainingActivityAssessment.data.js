import gql from 'graphql-tag';
export const TrainingActivityUpdateMutation = gql`
  mutation AddActivityAnswers(
    $sellerId: Int!
    $activityId: Int!
    $input: AddActivityAnswersInput!
    $ciclo: Int!
    $roleId: Int
    $grupo: Int!
    $gerenciaDeVendas: Int!
    $regiao: Int!
    $setor: Int!
    $gerenciaMercado: Int!
    $papelDaConsultora: Int!
    $canal: Int!
    $origem: String!
  ) {
    addActivityAnswers(
      sellerId: $sellerId
      activityId: $activityId
      input: $input
      ciclo: $ciclo
      roleId: $roleId
      grupo: $grupo
      gerenciaDeVendas: $gerenciaDeVendas
      gerenciaMercado: $gerenciaMercado
      regiao: $regiao
      setor: $setor
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      origem: $origem
    ) {
      action
      stoppedAt
    }
  }
`;
