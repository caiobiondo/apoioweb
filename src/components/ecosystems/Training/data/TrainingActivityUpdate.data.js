import gql from 'graphql-tag';

export const TrainingActivityUpdateMutation = gql`
  mutation UpdateActivity(
    $sellerId: Int!
    $activityId: Int!
    $input: UpdateActivityInput!
    $ciclo: Int!
    $roleId: Int
    $grupo: Int!
    $gerenciaDeVendas: Int!
    $regiao: Int!
    $setor: Int!
    $gerenciaMercado: Int!
    $papelDaConsultora: Int!
    $canal: Int!
    $origem: String!
  ) {
    updateActivity(
      sellerId: $sellerId
      activityId: $activityId
      input: $input
      ciclo: $ciclo
      roleId: $roleId
      grupo: $grupo
      gerenciaDeVendas: $gerenciaDeVendas
      gerenciaMercado: $gerenciaMercado
      regiao: $regiao
      setor: $setor
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      origem: $origem
    ) {
      status
      message
    }
  }
`;
