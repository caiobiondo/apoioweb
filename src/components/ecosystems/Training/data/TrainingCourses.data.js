import gql from 'graphql-tag';
import { Origem } from 'config';

const ITEMS_PER_PAGE = 20;

export const TrainingCoursesQuery = gql`
  query TrainingCoursesQuery(
    $sellerId: Int!
    $offset: Int!
    $limit: Int!
    $status: String
    $favorite: Boolean
    $filter: String
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
    $origem: String!
  ) {
    courses(
      sellerId: $sellerId
      offset: $offset
      limit: $limit
      status: $status
      favorite: $favorite
      filter: $filter
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
      origem: $origem
    ) {
      hasNextPage
      items {
        id
        accessToken
        categoryTitle
        clientIdentifier
        clientSecrets
        courseContent {
          video
          html5
          web
        }
        dateUpload
        durationInSeconds
        isfavorite
        ratedByYou
        status
        stoppedAt
        thumbnail
        title
        type
        views
      }
    }
  }
`;

export const updateQuery = (previousResult, { fetchMoreResult }) => {
  if (!fetchMoreResult) {
    return previousResult;
  }

  return Object.assign({}, previousResult, {
    courses: {
      __typename: previousResult.courses.__typename,
      hasNextPage: fetchMoreResult.courses.hasNextPage,
      items: [...previousResult.courses.items, ...fetchMoreResult.courses.items],
    },
  });
};

export const TrainingCoursesQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        limit: ITEMS_PER_PAGE,
        offset: 0,
        status: props.status,
        favorite: props.favorite,
        filter: props.courseFilter,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.codigoTipo > 3 ? props.user.estrutura.setor.codigo : 0,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigoTipo > 4 ? props.user.estrutura.codigo : 0,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas:
          props.user.estrutura.codigoTipo > 2 ? props.user.estrutura.gerenciaVenda.codigo : 0,
        regiao:
          props.user.estrutura.codigoTipo > 1 ? props.user.estrutura.regiaoEstrategica.codigo : 0,
        origem: Origem,
      },
      fetchPolicy: 'cache-first',
    };
  },
  props({ data }) {
    const { refetch, loading } = data;
    const courses = data.courses && data.courses.items;
    const hasNextPage = (data.courses && data.courses.hasNextPage) || false;

    return {
      data,
      refetch,
      loading,
      courses,
      hasNextPage,
      fetchMore() {
        if (data.loading) {
          return;
        }

        const offset = courses ? courses.length : 0;
        const variables = { offset };

        return data.fetchMore({
          variables,
          updateQuery,
        });
      },
    };
  },
};
