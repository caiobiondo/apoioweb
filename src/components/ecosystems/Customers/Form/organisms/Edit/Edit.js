import CustomerForm from '../Form';
import { UpdateCustomerMutation, FindCustomerQuery, FindCustomerQueryOptions } from './Edit.data';
import { graphql } from 'react-apollo';
import { withRouter } from 'react-router-dom';
import { withFormik } from 'formik';
import validateForm from '../../../Validators/Form';

class EditCustomerForm extends CustomerForm {
  render() {
    if (this.props.loading || !this.props.customer) {
      return null;
    }

    return super.render();
  }
}

const EditCustomerFormWithFormik = withFormik({
  mapPropsToValues: props => {
    if (!props.customer) {
      return { customer: null };
    }

    const addresses = props.customer.addresses || [];
    const phones = props.customer.phones || [];
    const emails = props.customer.emails || [];
    return {
      customer: {
        ...props.customer,
        addresses: addresses.map(address => {
          return { ...address };
        }),
        phones: phones.map(phone => {
          return { ...phone };
        }),
        emails: emails.map(email => {
          return { ...email };
        }),
      },
      editMode: true,
    };
  },
  enableReinitialize: true,
  validate: values => {
    return {
      customer: validateForm(values.customer),
    };
  },
})(EditCustomerForm);

const EditCustomerFormWithMutation = graphql(UpdateCustomerMutation)(EditCustomerFormWithFormik);

const EditCustomerFormWithData = graphql(FindCustomerQuery, FindCustomerQueryOptions)(
  EditCustomerFormWithMutation,
);
export { EditCustomerFormWithFormik as EditCustomerForm };
export default withRouter(EditCustomerFormWithData);
