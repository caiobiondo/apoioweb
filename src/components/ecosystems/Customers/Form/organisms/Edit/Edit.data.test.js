import { UpdateCustomerMutation, FindCustomerQuery, FindCustomerQueryOptions } from './Edit.data';

describe('UpdateCustomerMutation', () => {
  it('should be the correct mutation', () => {
    expect(UpdateCustomerMutation).toMatchSnapshot();
  });
});

describe('FindCustomerQuery', () => {
  it('should be the correct query', () => {
    expect(FindCustomerQuery).toMatchSnapshot();
  });

  it('should be the correct query options', () => {
    const props = {
      match: {
        params: {
          id: 'id',
        },
      },
    };

    const options = FindCustomerQueryOptions.options(props);

    expect(options).toEqual({
      variables: {
        customerId: 'id',
      },
      forceFetch: true,
      fetchPolicy: 'cache-and-network',
    });
  });

  it('should be the correct query props', () => {
    const data = {
      data: {
        loading: true,
        customer: {
          id: 'id',
          prop: 'value',
        },
      },
    };

    const props = FindCustomerQueryOptions.props(data);

    expect(props).toEqual({
      loading: true,
      customer: {
        id: 'id',
        prop: 'value',
      },
    });
  });
});
