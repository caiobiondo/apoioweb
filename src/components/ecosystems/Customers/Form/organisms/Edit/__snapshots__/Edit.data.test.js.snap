// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindCustomerQuery should be the correct query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "FindCustomerQuery",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "customerId",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "customerId",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "customer",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "nickname",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "gender",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "comment",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "birthday",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "emails",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "email",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "phones",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "phone",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "provider",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "addresses",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "additional_address",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "city",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "neighborhood",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "street_name",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "street_number",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "state",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "zipcode",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "customerId",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 425,
    "start": 0,
  },
}
`;

exports[`UpdateCustomerMutation should be the correct mutation 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "UpdateCustomer",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "input",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "input",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "updateCustomer",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "customer",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "nickname",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "UpdateCustomerInput",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "input",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 171,
    "start": 0,
  },
}
`;
